orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/adapters/RetryableTaskTasklet.groovy

```diff
@@ -72,4 +89,11 @@ class RetryableTaskTasklet extends TaskTasklet {
     }
     return super.doExecuteTask(stage, chunkContext)
   }
+
+  /**
+   * In order to accurately determine whether this task has timed out, any time spent in a paused state must be discounted.
+   */
+  static long determineNow(Clock clock, PausedDetails pausedDetails) {
+    return clock.millis() - ((pausedDetails?.pausedMs ?: 0) as Long)
```

Comment:
For example, if we were paused for 30 minutes ... I want to subtract 30 minutes from clock.millis() when determining whether or not the stage timeout has been exceeded.